package-res/resources/web/pentaho/util/arg.js

```diff
@@ -20,23 +20,73 @@ define([
 
   var A_slice = Array.prototype.slice;
 
+  /**
+   * @name pentaho.util.arg
+   * @class
+   * @amd pentaho/util/arg
+   *
+   * @classDesc Util methods for keyword args.
+   *
+   * The AMD module returns the arg utils.
+   *
+   * @description Performs actions on keyword args.
+   */
   return {
+    /**
+     * Provides the optional property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
+     * @return {any} The found optional property value or the default value.
+     */
     optional: function(o, p, dv) {
       var v;
       return o && (v = o[p]) != null ? v : dv;
     },
 
+    /**
+     * Provides the defined property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
+     * @return {any} The found defined property value (a property is considered defined if its value is not undefined) or the default value.
+     */
```

Comment:
what about "Returns the value of the property. If the property does not exist, returns `defaultValue`." ?